wdl-gauntlet/src/report.rs

```diff
@@ -36,7 +36,7 @@ pub enum Status {
     /// The document had diagnostics, but the diagnostics exactly matched what
     /// was already expected in the configuration, meaning that this
     /// [`Status`] is considered successful.
-    DiagnosticsMatched(IndexSet<String>),
+    DiagnosticsMatched(IndexSet<(Option<usize>, String)>),
```

Comment:
Likewise, this could probably just mirror the `all` above by being an `IndexMap<String, Option<usize>>`.